src/rust/lib/graph-editor/src/component/visualization.rs

```diff
@@ -0,0 +1,332 @@
+//! This module defines the visualization widgets and related functionality.
+//!
+//! At the core of this functionality is the `Visualisation` that takes in data and renders an
+//! output visualisation which is displayed in a `Container`. The `Container` provides generic UI
+//! elements that facilitate generic interactions, for example, visualisation selection. The
+//! `Container` also provides the FRP API that allows internal interaction with the
+//! `Visualisation`. Data for a visualisation has to be provided wrapped in the `Data` struct.
+//!
+use crate::prelude::*;
+
+use crate::frp;
+
+use ensogl::display::DomSymbol;
+use ensogl::display::object::class::Object;
+use ensogl::display::object::class::ObjectOps;
+use ensogl::display;
+use ensogl::system::web;
+use web::StyleSetter;
+
+
+// ============================================
+// === Wrapper for Visualisation Input Data ===
+// ============================================
+
+/// Wrapper for data that can be consumed by a visualisation.
+/// TODO replace with better typed data wrapper.
+#[derive(Clone,Debug)]
+#[allow(missing_docs)]
+pub enum Data {
+    JSON { content : String },
+    Empty,
```

Comment:
I know there is TODO for it, but just a note. If in any case you need a data which can be empty, always use Option :)